frontend/src/pages/Profile/index.tsx

```diff
@@ -0,0 +1,87 @@
+import { useParams } from "react-router-dom";
+import AppMargin from "../../components/AppMargin";
+import ProfileSection from "../../components/ProfileSection";
+import { Box, Typography } from "@mui/material";
+import classes from "./index.module.css";
+import { useEffect, useState } from "react";
+import { userClient } from "../../utils/api";
+import { useAuth } from "../../contexts/AuthContext";
+
+type UserProfile = {
+  id: string;
+  username: string;
+  firstName: string;
+  lastName: string;
+  email: string;
+  isAdmin: boolean;
+  biography?: string;
+  profilePictureUrl?: string;
+  createdAt: string;
+};
+
+const ProfilePage: React.FC = () => {
+  const { userId } = useParams<{ userId: string }>();
+  const [userProfile, setUserProfile] = useState<UserProfile | null>(null);
+  const auth = useAuth();
+
+  if (!auth) {
+    throw new Error("useAuth() must be used within AuthProvider");
+  }
+
+  const { user } = auth;
+
+  useEffect(() => {
+    userClient
+      .get(`/users/${userId}`)
+      .then((res) => {
+        setUserProfile(res.data.data);
+      })
+      .catch(() => setUserProfile(null));
+  }, []);
```

Comment:
Maybe include the `userId` in the dependency array here? So that `useEffect` will run when `userId` changes